### PR TITLE
Add fuzzing audit report

### PR DESCRIPTION
**Pull request by AdamKorcz**
Adds report for the fuzzing audit performed by Ada Logics.

@DavidKorczynski @spzala @caniszczyk

### CODE REVIEW

**Comment by ptabor** [comment]
Thank you @AdamKorcz. 
@nate-double-u, @spzala  - do you have opinion whether we should keep such *.pdf reports in the source-code repository or website. My intuition is that it should be the webpage repo and we should move both the reports there. WDYT ?

**Comment by AdamKorcz** [comment]
> > Thank you @AdamKorcz. @nate-double-u, @spzala - do you have an opinion on whether we should keep such *.pdf reports in the source-code repository or website. My intuition is that it should be the webpage repo and we should move both the reports there. WDYT ?
> 
> Hi @ptabor - yes, we should keep content in there but considering we have a dedicated `security` folder here and that since the etcd.io/etcd repo is more visible, I think it's okay to keep such findings here but your suggestion is good too. Let's keep the new report here for now, and if we decide to move reports to the `website` repo, I will work on moving both reports there. If that sound good? Thanks!

SGTM. There may be the issue that the existing security audit report is linked to from around the internet, and those links will then result in a 404 if it gets moved.

File: security/README.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ A public disclosure date is negotiated by the etcd Product Security Committee an
 ## Security Audit
 
 A third party security audit was performed by Trail of Bits, find the full report [here](SECURITY_AUDIT.pdf).
+A third fuzzing security audit was performed by Ada Logics, find the full report [here](FUZZING_AUDIT_2022.pdf).
```
Comment (spzala) [inline security/README.md:38]: @AdamKorcz a small catch - `third party` :)

Comment (AdamKorcz) [reply security/README.md]: Thanks for catching that!

**Comment by spzala** [comment]
Thanks again for the great work @AdamKorcz  Merging this per discussion with @ptabor We will think of a place on the Website repo to move current reports and link them from etcd, and also use Website for any future findings/reports. @nate-double-u thanks for your review.

**Pull request merged**